packages/frint-router-react/src/Link.spec.js

```diff
@@ -0,0 +1,264 @@
+/* eslint-disable import/no-extraneous-dependencies, func-names */
+/* global describe, it, before, resetDOM */
+import { expect } from 'chai';
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+import { MemoryRouterService } from 'frint-router';
+
+import Link from './Link';
+
+function createContextWithRouter(router) {
+  return {
+    app: {
+      get(key) {
+        if (key === 'router') {
+          return router;
+        }
+
+        return null;
+      }
+    }
+  };
+}
+
+describe('frint-route-react › Link', () => {
+  before(function () {
+    resetDOM();
+  });
+
+  it('renders anchor with href, className and children when no type prop is passed', () => {
+    const linkContent = <span className="decorated">About</span>;
+
+    const wrapper = shallow(
+      <Link className="anchor" to="/about">{linkContent}</Link>,
+      { context: createContextWithRouter() }
+    );
+
+    expect(wrapper.name()).to.equal('a');
+    expect(wrapper.hasClass('anchor')).to.be.true; // eslint-disable-line no-unused-expressions
+
+    expect(wrapper.children().length).to.equal(1);
+    expect(wrapper.children().get(0)).to.equal(linkContent);
+  });
+
+  it('renders button with type, className and children when type prop is passed', () => {
+    const linkContent = <span className="decorated">About</span>;
+
+    const wrapper = shallow(
+      <Link className="fancy-button" to="/about" type="button">{linkContent}</Link>,
+      { context: { app: {} } }
+    );
+
+    expect(wrapper.name()).to.equal('button');
+    expect(wrapper.prop('type')).to.equal('button');
+    expect(wrapper.hasClass('fancy-button')).to.be.true; // eslint-disable-line no-unused-expressions
+
+    expect(wrapper.children().length).to.equal(1);
+    expect(wrapper.children().get(0)).to.equal(linkContent);
+  });
+
+  it('subscribes to router service for  matching route inexactly and changes activeClass accordingly ' +
+    'when activeClassName is passed', () => {
+    const router = new MemoryRouterService();
+
+    const wrapper = mount(
+      <Link
+        activeClassName="anchor-active"
+        className="anchor"
+        to="/about"
+      >
+        About
+      </Link>,
+      { context: createContextWithRouter(router) }
+    );
+
+    router.push('/');
+    expect(wrapper.hasClass('anchor')).to.be.true; // eslint-disable-line no-unused-expressions
+    expect(wrapper.hasClass('anchor-active')).to.be.false; // eslint-disable-line no-unused-expressions
+
+    router.push('/about');
+    expect(wrapper.hasClass('anchor')).to.be.true; // eslint-disable-line no-unused-expressions
```

Comment:
since too many similar inline eslint disable comments, consider moving it to the top of the file.